opencti-platform/opencti-graphql/src/utils/access.ts

```diff
@@ -211,3 +216,57 @@ export const isUserCanAccessStixElement = async (context: AuthContext, user: Aut
   // If no platform organization is set, user can access empty sharing and dedicated sharing
   return elementOrganizations.length === 0 || elementOrganizations.some((r) => userOrganizations.includes(r));
 };
+
+// region member access
+
+// returns all user member access ids : his id, his organizations ids (and parent organizations), his groups ids
+export const computeUserMemberAccessIds = (user: AuthUser) => {
+  const memberAccessIds = [user.id];
+  if (user.allowed_organizations) {
+    const userOrganizationsIds = user.allowed_organizations.map((org) => org.internal_id);
+    memberAccessIds.push(...userOrganizationsIds);
+  }
+  if (user.groups) {
+    const userGroupsIds = user.groups.map((group) => group.internal_id);
+    memberAccessIds.push(...userGroupsIds);
+  }
+  return memberAccessIds;
+};
+
+// user access methods
+export const getUserAccessRight = (user: AuthUser, element: any) => {
+  if (isBypassUser(user) || !element.authorized_members) { // no restricted user access on element
+    return MEMBER_ACCESS_RIGHT_ADMIN;
+  }
+  const accessMembers = [...element.authorized_members];
+  const userMemberAccessIds = computeUserMemberAccessIds(user);
+  const foundAccessMembers = accessMembers.filter((u) => u.id === 'ALL' || userMemberAccessIds.includes(u.id));
+  if (!foundAccessMembers.length) { // user has no access
+    return null;
+  }
+  if (foundAccessMembers.some((m) => m.access_right === MEMBER_ACCESS_RIGHT_ADMIN)) {
+    return MEMBER_ACCESS_RIGHT_ADMIN;
+  }
+  if (foundAccessMembers.some((m) => m.access_right === MEMBER_ACCESS_RIGHT_EDIT)) {
+    return MEMBER_ACCESS_RIGHT_EDIT;
+  }
+  return MEMBER_ACCESS_RIGHT_VIEW;
```

Comment:
This method `getUserAccessRight` returns the appropriate access right for a user (because a user could have multiple access rights through his groups, organizations or ALL) : so we return first admin if present, then edit if present, then by default it will be view access.